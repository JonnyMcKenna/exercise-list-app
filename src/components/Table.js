import "../App.css";
import TableData from "./TableData";

const Table = (props) => {
  const { exercises, gender } = props;

  return (
    exercises.length !== 0 && (
      <table className="exercise-data">
        {exercises.map(function (exercise, index) {
          return <TableData key={index} exercise={exercise} gender={gender} />;
        })}
      </table>
    )
  );
};

export default Table;
