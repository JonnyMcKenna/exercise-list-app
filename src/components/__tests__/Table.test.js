import React from "react";
import Table from "../Table";
import { render } from "@testing-library/react";

describe("Table", () => {
  it("should render when exercises array is not empty", () => {
    const exercises = [
      {
        id: "123",
        name: "name",
        transcript: "transcript",
        male: { image: "image" },
        female: { image: "image" },
      },
    ];
    const appElement = render(<Table exercises={exercises} />).baseElement;
    expect(appElement).toMatchSnapshot();
    expect(appElement).toHaveTextContent("View Instructions");
  });

  it("should not render when exercises array is empty", () => {
    const exercises = [];
    const appElement = render(<Table exercises={exercises} />).baseElement;
    expect(appElement).toMatchSnapshot();
  });
});
