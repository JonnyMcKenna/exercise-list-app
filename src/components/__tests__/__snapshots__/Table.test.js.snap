// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should not render when exercises array is empty 1`] = `
<body>
  <div />
</body>
`;

exports[`Table should render when exercises array is not empty 1`] = `
<body>
  <div>
    <table
      class="exercise-data"
    >
      <tbody>
        <tr
          class="tableRow"
        >
          <td
            class="firstColumn"
          >
            <div
              class="exerciseName"
            >
              name
            </div>
            <div />
            <button
              class="view-instructions-link"
              id="view-instructions-link-id"
            >
              View Instructions
            </button>
          </td>
          <td
            class="secondColumn"
          >
            <div />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
